pkg/pool-weighted/test/foundry/SeedlessLBPPool.t.sol

```diff
@@ -835,10 +615,179 @@ contract SeedlessLBPTest is WeightedLBPTest {
         uint256 projectReceived = projectToken.balanceOf(bob) - projectBefore;
 
         // Owner should get all the accumulated reserve (proceeds) and remaining project tokens
-        assertApproxEqRel(reserveReceived, totalBought, 0.01e18, "Owner should receive sale proceeds");
+        assertApproxEqRel(reserveReceived, totalBought, 1e3, "Owner should receive sale proceeds");
         assertGt(projectReceived, 0, "Owner should receive remaining project tokens");
     }
 
+    struct SwapTestLocals {
+        address seedlessPool;
+        address seededPool;
+        IERC20 projToken;
+        IERC20 resToken;
+        uint256 projIdx;
+        uint256 resIdx;
+        uint256 virtualBalance;
+        uint256 projectInitAmount;
+    }
+
+    function testSeedlessVsSeededSwaps__Fuzz(uint256 swapAmount, uint256 timeOffset, bool useNon18Decimals) public {
```

Comment:
Good stuff.

Maybe one small suggestion here: I think it'd be better to have 2 simpler tests as opposed to one complex test. In particular, I'd remove the `useNon18Decimals` argument, and just do one test for 18-decimal tokens and another one for non-18.

The code will be mostly the same and you can even factor it out to use the same underlying functions after initialization, but you save all the `if / else`s in the test, you'll get more coverage and it'll be simpler to read.